src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs

```diff
@@ -332,28 +344,40 @@ impl NonOverlapSubLevelPicker {
         &self,
         l0: &[Level],
         level_handler: &LevelHandler,
+        stats: &mut LocalPickerStatistic,
     ) -> Vec<SubLevelSstables> {
         if l0.len() < self.min_depth {
             return vec![];
         }
 
         let mut scores = vec![];
         let select_tables = &l0[0].table_infos;
+        let mut skip_by_count = false;
+        let mut skip_by_pending = false;
         for (sst_index, sst) in select_tables.iter().enumerate() {
             if level_handler.is_pending_compact(&sst.sst_id) {
+                skip_by_pending = true;
                 continue;
             }
 
             let ret = self.pick_sub_level(l0, level_handler, sst_index, sst);
             if ret.sstable_infos.len() < self.min_depth
                 && ret.total_file_size < self.min_compaction_bytes
             {
+                skip_by_count = true;
                 continue;
             }
             scores.push(ret);
         }
 
         if scores.is_empty() {
```

Comment:
Why only update the skip stat on empty score?

Reply:
to clarify why a task is not selected, when all tasks are limited by min_depth/ min_compaction_bytes.